build.cake

```diff
@@ -0,0 +1,268 @@
+
+// Solutions
+var ADAPTER_SOLUTION = PROJECT_DIR + "NUnit3TestAdapter.sln";
+var DEMO_SOLUTION = DEMO_DIR + "NUnit3TestDemo.sln";
+
+// Test Runners
+var NUNIT3_CONSOLE = TOOLS_DIR + "NUnit.ConsoleRunner/tools/nunit3-console.exe";
+
+// Test Assemblies
+var ADAPTER_TESTS = TEST_BIN_DIR + "NUnit.VisualStudio.TestAdapter.Tests.dll";
+var DEMO_TESTS = DEMO_BIN_DIR + "NUnit3TestDemo.dll";
+
+// Packages
+var ZIP_PACKAGE = PACKAGE_DIR + "NUnit3TestAdapter-" + packageVersion + ".zip";
+
+// Custom settings for VSTest
+var VSTestCustomSettings = new VSTestSettings()
+{
+	ArgumentCustomization = args => args.Append("/TestAdapterPath:" + ADAPTER_BIN_DIR)
+};
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectory(ADAPTER_BIN_DIR);
+    CleanDirectory(TEST_BIN_DIR);
+    CleanDirectory(INSTALL_BIN_DIR);
+	CleanDirectory(DEMO_BIN_DIR);
+});
+
+
+//////////////////////////////////////////////////////////////////////
+// INITIALIZE FOR BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("InitializeBuild")
+    .Does(() =>
+{
+    NuGetRestore(ADAPTER_SOLUTION);
+	NuGetRestore(DEMO_SOLUTION);
+
+	if (BuildSystem.IsRunningOnAppVeyor)
+	{
+		var tag = AppVeyor.Environment.Repository.Tag;
+
+		if (tag.IsTag)
+		{
+			packageVersion = tag.Name;
+		}
+		else
+		{
+			var buildNumber = AppVeyor.Environment.Build.Number;
+			packageVersion = version + "-CI-" + buildNumber + dbgSuffix;
+			if (AppVeyor.Environment.PullRequest.IsPullRequest)
+				packageVersion += "-PR-" + AppVeyor.Environment.PullRequest.Number;
+			else
+				packageVersion += "-" + AppVeyor.Environment.Repository.Branch;
+		}
+
+		AppVeyor.UpdateBuildVersion(packageVersion);
+	}
+});
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+    .IsDependentOn("InitializeBuild")
+    .Does(() =>
+    {
+		BuildSolution(ADAPTER_SOLUTION, configuration);
+		BuildSolution(DEMO_SOLUTION, configuration);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("TestAdapterUsingConsole")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
+        StartProcess(
+			NUNIT3_CONSOLE,
+			new ProcessSettings()
+			{
+				Arguments = ADAPTER_TESTS
+			});
+	});
+
+Task("TestAdapterUsingVSTest")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
+		VSTest(ADAPTER_TESTS, VSTestCustomSettings);
+	});
+
+Task("RunTestDemo")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
+		try
+		{
+			VSTest(DEMO_BIN_DIR + "NUnit3TestDemo.dll", VSTestCustomSettings);
+		}
+		catch(Exception ex)
+		{
+			Information("\nNOTE: Demo tests failed as expected.");
+			Information("This is normally not an error.\n");
+		}
+	});
+
```

Comment:
I took a look at the options here. The trouble is that I don't really know anything about VSTest - this is the first time I ever used it. There doesn't seem to be an option to capture the output so I can scan for the expected result. Anyway, the actual number of failures will change with time anyway, so the only thing I could check is that it ran to completion and produced a summary line.

I also determined experimentally that an actual error, like putting in an invalid option to the call, produces the same exception, return code and message as a test failure.

So, all in all, it seemed easiest to put in a special message at the end, allowing folks to look up a few lines if they are looking at the console. The alternative, of course, would be to take this out of the standard AppVeyor tests entirely, only running it locally by command.